__init__.py

```diff
@@ -0,0 +1,326 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Mycroft skill for reporting the current and future time.
+
+NOTE ON REGULAR EXPRESSION HANDLING:
+    Most skills that have regular expression files (those with a ".rx" extension) can
+    use the intent system to determine if words matching the regular expression(s) are
+    present in the utterance.  This has the benefit of improving intent confidence
+    calculations.  This skill, however, foregoes that mechanism.
+
+    Consider the following request from the user:
+        What time is it in London in 5 hours?
+
+    A regular expression to find the location in this request is difficult to write.
+    The location can be found looking of the word "in" followed by a city name.  But in
+    this case, the future offset "in 5 hours" would also match this type of pattern.
+
+    To get around this situation, this skill does not pass the regular expression on to
+    Adapt.  Instead, it pulls the future offset out of the user's request first,
+    leaving only the city to match the regular expression looking for a location.
+"""
+import time
+from pathlib import Path
+
+from mycroft.audio import wait_while_speaking
+from mycroft.messagebus.message import Message
+from mycroft.skills import MycroftSkill, intent_handler
+from mycroft.skills.intent_service import AdaptIntent
+from mycroft.util.format import date_time_format
+from mycroft.util.time import now_local
+from .skill import FaceplateRenderer, get_display_time, LocationNotFoundError, Response
+
+MARK_I = "mycroft_mark_1"
+MARK_II = "mycroft_mark_2"
+
+
+class TimeSkill(MycroftSkill):
+    """Mycroft skill for reporting the current and future time."""
+
+    def __init__(self):
+        super().__init__("TimeSkill")
+        self.displayed_time = None
+        self.location_regex_path = Path(self.find_resource("location.rx"))
+
+    # TODO: define in the base class so it can be accessed by other skills.
+    @property
+    def platform(self):
+        """Get the platform identifier string
+
+        Returns:
+            str: Platform identifier, such as "mycroft_mark_1",
+                 "mycroft_picroft", "mycroft_mark_2".  None for non-standard.
+        """
+        platform = None
+        if self.config_core and self.config_core.get("enclosure"):
+            platform = self.config_core["enclosure"].get("platform")
+
+        return platform
+
+    # TODO: Move the Mark I functionality for idle screens into the Home Screen skill
+    @property
+    def display_when_idle(self):
+        """Syntactic sugar for whether or not the time should be displayed when idle."""
+        show_time_when_idle = False
+        if self.platform == MARK_I:
+            show_time_when_idle = self.settings.get("show_time", False)
+
+        return show_time_when_idle
+
+    def initialize(self):
+        """Do the things after the constructor bug before logic is executed."""
```

Comment:
Typo maybe should say 'constructor but' not 'constructor bug'. This is known as a Freudian slip.

Reply:
good catch.  fixed.